src/query/service/src/interpreters/interpreter_copy_v2.rs

```diff
@@ -263,61 +266,68 @@ impl CopyInterpreterV2 {
-                    CopyInterpreterV2::purge_files(ctx, &from, &files).await
-                });
+        if !files.is_empty() {
+            let mut build_res = PipelineBuildResult::create();
+            let read_source_plan = Self::rewrite_read_plan_file_name(from.clone(), &files);
+            tracing::info!("copy_files_to_table from source: {:?}", read_source_plan);
+
+            let from_table = self.ctx.build_table_from_source_plan(&read_source_plan)?;
+            from_table.read2(
+                self.ctx.clone(),
+                &read_source_plan,
+                &mut build_res.main_pipeline,
+            )?;
+
+            let to_table = self.ctx.get_table(catalog_name, db_name, tbl_name).await?;
+            to_table.append2(self.ctx.clone(), &mut build_res.main_pipeline, false)?;
+
+            let ctx = self.ctx.clone();
+            let catalog_name = catalog_name.clone();
+            let files = files.clone();
+            let from = from.clone();
+
+            build_res.main_pipeline.set_on_finished(move |may_error| {
+                if may_error.is_none() {
+                    // capture out variable
+                    let ctx = ctx.clone();
+                    let files = files.clone();
+                    let from = from.clone();
+                    let catalog_name = catalog_name.clone();
+                    let to_table = to_table.clone();
+
+                    let task = GlobalIORuntime::instance().spawn(async move {
+                        // Commit
+                        let operations = ctx.consume_precommit_blocks();
+                        to_table
+                            .commit_insertion(ctx.clone(), &catalog_name, operations, false)
+                            .await?;
 
-                return match futures::executor::block_on(task) {
-                    Ok(Ok(_)) => Ok(()),
-                    Ok(Err(error)) => Err(error),
-                    Err(cause) => Err(ErrorCode::PanicError(format!(
-                        "Maybe panic while in commit insert. {}",
-                        cause
-                    ))),
-                };
-            }
+                        // Purge
+                        CopyInterpreterV2::purge_files(ctx, &from, &files).await
+                    });
+
+                    return match futures::executor::block_on(task) {
+                        Ok(Ok(_)) => Ok(()),
+                        Ok(Err(error)) => Err(error),
+                        Err(cause) => Err(ErrorCode::PanicError(format!(
+                            "Maybe panic while in commit insert. {}",
+                            cause
+                        ))),
+                    };
+                }
 
-            Err(may_error.as_ref().unwrap().clone())
-        });
+                Err(may_error.as_ref().unwrap().clone())
+            });
+        }
 
-        Ok(build_res)
+        // Copy status.
+        let schema = Arc::new(DataSchema::new(vec![DataField::new(
```

Comment:
Please change `CopyPlanV2` into a `has_result` plan.

Reply:
Thank you.
I am working on why it does not return the result, you saved me 😢